static/scenes/railroad/js/present.js

```diff
@@ -20,38 +26,46 @@ class Present {
                 }
               }
             this.model = obj;
-            this.scene.getScene().add(obj);
-            console.log("Present added");
+            this.scene.camera.add(obj);
+            obj.position.set(.125, -.125, -1);
         });
     }
 
     shoot(targetPosition) {
-        this.targetPosition = targetPosition;
-        this.startPosition = this.model.position;
+        this.scene.scene.attach(this.model);
+        this.targetPosition = targetPosition.clone();
+        this.startPosition = new THREE.Vector3();
+        this.model.getWorldPosition(this.startPosition);
+
+        var midpoint = this.startPosition.clone()
+            .multiplyScalar(1-midpointT)
+            .addScaledVector(this.targetPosition, midpointT)
+            .add(new THREE.Vector3(0, this.startPosition.distanceTo(this.targetPosition) * midpointOffset));
+
+        this.curve = new THREE.QuadraticBezierCurve3(this.startPosition.clone(), midpoint.clone(), this.targetPosition.clone());
 
-        var midpoint = new THREE.Vector3().add(this.targetPosition, this.startPosition).multiplyScalar(0.5);
-        midpoint.setY(midpoint.y + 3);
-        this.curve = new THREE.QuadraticBezierCurve3(this.startPosition, midpoint, this.targetPosition);
         this.inFlight = true;
     }
 
+    addDebugGeo() {
+        const points = this.curve.getPoints(50);
+        const geometry = new THREE.BufferGeometry().setFromPoints(points);
+        const material = new THREE.LineBasicMaterial({color: 0x00ff00});
+        const curveObject = new THREE.Line(geometry, material);
+        this.scene.scene.add(curveObject);
+    }
+
     update(seconds, deltaSeconds) {
         if (this.inFlight) {
             if (this.currentFlightTime > this.totalFlightTime) {
                 this.landed = true;
+                this.model.position.copy(this.targetPosition);
```

Comment:
We could also just clamp the t value below, and have that code block run all the time.
https://threejs.org/docs/#api/en/math/MathUtils.clamp

Reply:
Either works, I like not running a whole interpolation and the relevant maths when we don't have to. And this guarantees that it will be at exactly the target position, which really doesn't matter in the current implementation.

Reply:
Sure. Interpolation is fast, but given it's a single operation per frame, it's really just a style thing at this stage. In some situations clamping + interpolation might be faster than using an if statement thanks to avoiding the branch